.github/workflows/c-cpp.yml

```diff
@@ -0,0 +1,172 @@
+name: C++ CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  cmake_test:
+    name: Build with CMake, then run tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: setup
+      run: |
+        sudo ./ci/setup_cmake.sh
+        sudo ./ci/setup_ci_environment.sh
+    - name: run tests
+      run: ./ci/do_ci.sh cmake.test
+
+  plugin_test:
+    name: Statically link a plugin library, then build using CMake
```

Comment:
Could we choose shorter names for the jobs? Otherwise it's hard to quickly identify jobs:

<img width="303" alt="image" src="https://user-images.githubusercontent.com/16337442/84553954-a5028b00-acca-11ea-93e9-0252a666449e.png">

Reply:
An earlier reviewer asked me to provide more descriptive names for the jobs to better differentiate some that are similar. I tried to keep it brief but still descriptive. But if you have any suggestions I would be happy to add them! 😄

Reply:
Sorry for not thinking of the implication on the UI, and wish there was a `job.description` page that showed up when clicking the job, but not on the left side...

Think most of the `Build with` can be dropped, and in general don't need to be sentences. "Static link plugin -> CMake", type of phrasing would still provide more than `plugin_test` and hopefully not overflow the UI.

Reply:
I see. I just tried to shorten the job names while still keeping them as descriptive as possible. Let me know what you think!

Reply:
Thanks!